ci/test/03_test_script.sh

```diff
@@ -144,21 +144,70 @@ if [ "$RUN_CHECK_DEPS" = "true" ]; then
   "${BASE_ROOT_DIR}/contrib/devtools/check-deps.sh" .
 fi
 
+function get_interfaces()
+{
+  ifconfig | awk -F : '/^[^[:space:]]/ { if ($1 != "lo") { print $1 } }'
+}
+
+function tcpdump_file_for_interface()
+{
+  echo "/tmp/tcpdump_$1"
+}
+
+function traffic_monitor_begin()
+{
+  for ifname in $(get_interfaces) ; do
+    tcpdump -n -i "$ifname" -w "$(tcpdump_file_for_interface "$ifname")" &
```

Comment:
On some of the VMs this produces:

```
[09:47:01.270] + tcpdump -n -i eth0 -w /tmp/tcpdump_eth0
[09:47:01.334] tcpdump: eth0: You don't have permission to perform this capture on that device
[09:47:01.335] (socket: Operation not permitted)
```

and then the CI passes because the return code is ignored. I think better not fail the CI when `tcpdump` does not work in that environment. It is ok as long as `tcpdump` works on at least one VM to catch problems.

Reply:
True, if it stops working on all VMs, then nobody will notice. Any ideas how to approach this?

Reply:
> (unless they are switched to @0xB10C's workers, which are running as root?)

the runner setup I'm working on explicitly **doesn't** run as root and is far from finished :)

Reply:
I'm using [rootless-docker](https://docs.docker.com/engine/security/rootless/) which runs dockerd as a user account. Inside the container, you're root and can tcpdump on the containers `eth0` interface, but you can't* e.g. mount and edit the hosts `/etc/passwd` like you can't with the user account. 

*until someone finds a vuln in rootless-docker

Reply:
> The problem is that no one will notice if this isn't run on any machine

I changed it to insist that the tcpdump file was created on the ASAN env. So the ASAN job will be red if this stops working on it.

Reply:
> ... explicitly list the required (or removed) caps ...

Done, `NET_RAW` is required to run `tcpdump`: https://www.tcpdump.org/manpages/pcap.3pcap.html

Plus, the ASAN job requires that `tcpdump -w` runs and creates the file, otherwise it will be red.

I think that resolves the concerns from this thread, so I am closing it. Feel free to comment / reopen if there is more to this.